da/mock/zone.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass
+from typing import List, Optional, Generator
+
+from da.common import Certificate, NodeId, Attestation
+from da.dispersal import Dispersal
+from da.encoder import EncodedData, DAEncoder, DAEncoderParams
+from da.verifier import Attestation, DABlob, DAVerifier
+
+from da.mock.common import CertificateMessage, DaApiMessage, Id, Metadata, data_hash
+from da.mock.node import MockDaNode
+from da.mock.producer import MockProducerNode
+from da.mock.transport import Transport
+
+@dataclass
+class MockZoneParams:
+    da_nodes: List[MockDaNode]
+    block_producer: MockProducerNode
+    threshold: int
+    encoder_params: DAEncoderParams
+
+class MockZoneNode:
```

Comment:
We can call this `Client` or `DaClient`. Is the better abstraction as not necessarily a *Zone* is gonna be the only thing to use this.